### PR TITLE
[now-next] Fix static 404 not being used in mono-repo set-up

### PR DESCRIPTION
**Pull request by ijjk**
This fixes the static 404 page not being used when deploying in a mono-repo structure. Before we weren't taking into account the `entryDirectory` where we needed to causing us to deploy `_error` when we didn't need to

x-ref: https://github.com/zeit/now/discussions/4077#discussioncomment-4625

### CODE REVIEW

File: packages/now-next/src/index.ts [hunk 1]
```diff
@@ -569,13 +569,13 @@ export const build = async ({
...
         // error handling
-        ...(output['404']
+        ...(output[path.join(entryDirectory, '404')]
           ? [
               { handle: 'error' } as Handler,
 
               {
                 status: 404,
-                src: path.join(entryDirectory, '.*'),
+                src: '/.*',
```
Comment (Timer) [inline packages/now-next/src/index.ts]: This still needs entry directory:
```suggestion
                src: path.join(entryDirectory, '.*'),
```

File: packages/now-next/src/index.ts [hunk 2]
```diff
@@ -569,13 +569,13 @@ export const build = async ({
         },
 
         // error handling
-        ...(output['404']
+        ...(output[path.join(entryDirectory, '404')]
```
Comment (Timer) [inline packages/now-next/src/index.ts]: Should this be this to be safe?
```suggestion
        ...(output[path.join('./', entryDirectory, '404')]
```

File: packages/now-next/src/index.ts [hunk 3]
```diff
@@ -1258,18 +1258,16 @@ export const build = async ({
...
-                entryDirectory,
-                static404Page
-                  ? static404Page
-                  : // if static 404 is not present but we have pages/404.js
-                  // it is a lambda due to _app getInitialProps
-                  hasPages404 && lambdas['404']
-                  ? '404'
-                  : '_error'
-              ),
+              src: '/.*',
```
Comment (Timer) [inline packages/now-next/src/index.ts]:
```suggestion
              src: path.join('/', entryDirectory, '.*'),
```

File: packages/now-next/src/index.ts (hunk 2 above)
Comment (ijjk) [reply packages/now-next/src/index.ts]: Outputs don't have starting slash

Comment (Timer) [reply packages/now-next/src/index.ts]: The above code would remove it if present:
```js
> path.join('./', '/abc')
'abc'
```

Comment (ijjk) [reply packages/now-next/src/index.ts]: Ah yeah that works, updated 👍

**Review by Timer — APPROVED** [review]

**Pull request merged**